src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -48,7 +57,8 @@ public static AnalysisResults serveImageText(URL url)
     ByteString imageBytes = readImageBytes(url);
 
     String rawText = retrieveText(imageBytes);
-    AnalysisResults results = new AnalysisResults(rawText);
+    Set<String> categories = categorizeText(rawText);
+    AnalysisResults results = new AnalysisResults(rawText, categories);
```

Comment:
This is common to both `serveImageText()` methods, so it could be extracted to a helper method that takes in a `ByteString` and returns the `AnalysisResults`.